packages/core/react/src/WalletProvider.tsx

```diff
@@ -49,6 +64,7 @@ export function WalletProvider({
     wallets: adapters,
     ...props
 }: WalletProviderProps) {
+    const { connection } = useConnection();
```

Comment:
Unfortunately this introduces a dependency on `ConnectionProvider` here, but let's be honest, it has _always_ been a soft dependency since you can very nearly accomplish nothing transaction related unless you have a `ConnectionProvider`.

Reply:
In the event that a dapp passes a `connection` arg to `sendTransaction` that is not the one from `useConnection`, what should happen?

Reply:
Also, if the dapp is configured without `ConnectionProvider`, should we do anything here?

Reply:
> In the event that a dapp passes a connection arg to sendTransaction that is not the one from useConnection, what should happen?

I think you're hosed in this case. If that connection is for a different cluster than the cluster for which you've sought an MWA authorization (ie. an `auth_token`) then it's just not gonna work. You can supply different endpoints all the live-long day, so long as they resolve to the same `Cluster` though.

Folks who really need to go ham with customization here should supply their own `new SolanaMobileWalletAdapter({/* ... */})`.

> …if the dapp is configured without ConnectionProvider, should we do anything here?

In that case then `connection?.rpcEndpoint` will be `undefined` and the inference module will just return `mainnet-beta` all the time.